src/redisearch_rs/trie_rs/tests/integration/iter/prefixes.rs

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+use trie_rs::TrieMap;
+
+/// Return all the keys that are a prefix of the given target.
+fn prefixes<Data: Clone>(trie: &TrieMap<Data>, target: &[u8]) -> Vec<Data> {
+    trie.prefixes_iter(target).map(|v| v.to_owned()).collect()
+}
+
+#[test]
+fn empty_is_a_prefix_of_itself() {
+    let mut trie = TrieMap::new();
+    trie.insert(b"", b"".to_vec());
+    trie.insert(b"apple", b"apple".into());
+
+    assert_eq!(prefixes(&trie, b""), vec![b""]);
+}
+
+#[test]
+fn non_empty_prefixes() {
+    let mut trie = TrieMap::new();
+    trie.insert(b"apple", b"apple".to_vec());
```

Comment:
for readability of the goal, I would maybe have a different value than the key? Something like "apple-value" or sthg like this?

Reply:
It would make it more clear to see if `key`or `value` is extracted?

Reply:
Yes yes I know is clear, but IMO is more straightforward to clarify with the value. But I guess there is already a pattern on other tests so it's okey